eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/sorted/immutable/ImmutableSortedSetFromBagAdapter.java

```diff
@@ -0,0 +1,155 @@
+import org.eclipse.collections.impl.utility.Iterate;
+import org.eclipse.collections.impl.utility.internal.IterableIterate;
+import org.eclipse.collections.impl.utility.internal.SortedSetIterables;
+
+/**
+ * This class provides a {@link ImmutableSet} wrapper around any Bag interface instance.
+ */
+public final class ImmutableSortedSetFromBagAdapter<T> extends AbstractImmutableSortedSet<T>
+{
+    private final SortedBag<T> delegate;
+
+    public ImmutableSortedSetFromBagAdapter(SortedBag<T> delegate)
+    {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public ImmutableSortedSet<T> take(int count)
+    {
+        return SortedSets.immutable.ofAll(this.comparator(), this.delegate.distinctView().asLazy().take(count));
+    }
+
+    @Override
+    public ImmutableSortedSet<T> drop(int count)
+    {
+        return SortedSets.immutable.ofAll(this.comparator(), this.delegate.distinctView().asLazy().drop(count));
+    }
+
+    @Override
+    public Comparator<? super T> comparator()
+    {
+        return this.delegate.comparator();
+    }
+
+    @Override
+    public int size()
+    {
+        return this.delegate.sizeDistinct();
+    }
+
+    @Override
+    public void each(Procedure<? super T> procedure)
+    {
+        this.delegate.distinctView().each(procedure);
+    }
+
+    @Override
+    public Iterator<T> iterator()
+    {
+        return this.delegate.distinctView().iterator();
+    }
+
+    @Override
+    public int indexOf(Object object)
+    {
+        return Iterate.indexOf(this.delegate.distinctView(), object);
+    }
+
+    @Override
+    public void forEach(int from, int to, Procedure<? super T> procedure)
+    {
+        IterableIterate.forEach(this.delegate.distinctView(), from, to, procedure);
+    }
+
+    @Override
+    public void forEachWithIndex(int from, int to, ObjectIntProcedure<? super T> objectIntProcedure)
+    {
+        IterableIterate.forEachWithIndex(this.delegate.distinctView(), from, to, objectIntProcedure);
+    }
+
+    @Override
+    public T first()
+    {
+        return this.delegate.getFirst();
+    }
+
+    @Override
+    public T last()
+    {
+        return this.delegate.getLast();
+    }
+
+    @Override
+    public int compareTo(SortedSetIterable<T> o)
+    {
+        return SortedSetIterables.compare(this, o);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj == this)
+        {
+            return true;
+        }
+
+        if (!(obj instanceof Set))
+        {
+            return false;
+        }
+        Set<?> otherSet = (Set<?>) obj;
+        if (otherSet.size() != this.size())
+        {
+            return false;
+        }
+        try
+        {
+            return this.containsAll(otherSet);
+        }
+        catch (ClassCastException ignored)
+        {
+            return false;
+        }
+        catch (NullPointerException ignored)
+        {
+            return false;
```

Comment:
Are these `return false` lines covered by tests?

Reply:
@motlin I will revert back to an implementation similar to the rest